test/Royalties.t.sol

```diff
@@ -0,0 +1,792 @@
+        uint256 teamReceived      = postBalTeam - preBalTeam;
+        uint256 devReceived       = postBalDev - preBalDev;
+        uint256 deadReceived      = postBalDead - preBalDead;
+
+        // Verify amount received is amount sent.
+        assertEq(marketingReceived, gogeToken.royaltiesSent(1));
+        assertEq(teamReceived,      gogeToken.royaltiesSent(3));
+        assertEq(devReceived,       gogeToken.royaltiesSent(2));
+        assertEq(devReceived,       0);
+
+        // log amount
+        emit log_named_uint("marketing", gogeToken.royaltiesSent(1));
+        emit log_named_uint("dev",       gogeToken.royaltiesSent(2));
+        emit log_named_uint("team",      gogeToken.royaltiesSent(3));
+        emit log_named_uint("buyback",   gogeToken.royaltiesSent(4));
+        emit log_named_uint("cake",      gogeToken.royaltiesSent(5));
+    }
+
+    function test_royaltyTesting_getCirculatingMinusReserve() public {
+
+        // 100_000_000_000 ether -> total supply
+        //   5_000_000_000 ether -> used for liquidity
+
+        // Verify circulating supply is 95B tokens.
+        assertEq(ERC20(address(gogeToken)).getCirculatingMinusReserve(), 95_000_000_000 ether);
+
+        // send 5B tokens to 0 address.
+        gogeToken.transfer(address(gogeToken.deadAddress()), 5_000_000_000 ether);
+
+        // Verify circulating supply is 90B tokens.
+        assertEq(ERC20(address(gogeToken)).getCirculatingMinusReserve(), 90_000_000_000 ether);
+
+        // Add address(69) to excluded list and transfer address(69) tokens.
+        gogeToken.transfer(address(69), 2_000_000_000 ether);
+        gogeToken.excludeFromCirculatingSupply(address(69), true);
+
+        // Verify circulating supply is 88B tokens.
+        assertEq(ERC20(address(gogeToken)).getCirculatingMinusReserve(), 88_000_000_000 ether);
+
+        // Remove address(69) from excluded list.
+        gogeToken.excludeFromCirculatingSupply(address(69), false);
+
+        // Verify circulating supply is 88B tokens.
+        assertEq(ERC20(address(gogeToken)).getCirculatingMinusReserve(), 90_000_000_000 ether);
+
+        emit log_named_uint("circulating supply minus reserve", ERC20(address(gogeToken)).getCirculatingMinusReserve());
+        emit log_named_uint("total supply", ERC20(address(gogeToken)).totalSupply());
+        emit log_named_uint("uniswap balance", ERC20(address(gogeToken)).balanceOf(gogeToken.uniswapV2Pair()));
+        emit log_named_uint("dead balance", ERC20(address(gogeToken)).balanceOf(address(0)));
+        emit log_named_uint("dead balance", ERC20(address(gogeToken)).balanceOf(gogeToken.deadAddress()));
+
+        // Add address(69) to the excluded list.
+        gogeToken.excludeFromCirculatingSupply(address(69), true);
+
+        // Verify excludedFromCirculatingSupply is updating accordingly.
+        (bool _isExcludedPair, uint8 pos1) = gogeToken.isExcludedFromCirculatingSupply(gogeToken.uniswapV2Pair());
```

Comment:
Why would you exclude a pair from the circulating supply?

Reply:
I was just testing adding a pool of tokens and ensuring if excluded; it results in the correct circulating supply